sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py

```diff
@@ -0,0 +1,275 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import typing
+
+import opentelemetry.trace as trace
+from opentelemetry.context import Context
+from opentelemetry.trace.propagation.textmap import (
+    Getter,
+    Setter,
+    TextMapPropagator,
+    TextMapPropagatorT,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class AwsParseTraceHeaderError(Exception):
+    def __init__(self, message):
+        super().__init__()
+        self.message = message
+
+
+class AwsXRayFormat(TextMapPropagator):
+    """Propagator for the AWS X-Ray Trace Header propagation protocol.
+
+    See: https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader
+    """
+
+    # AWS
+    TRACE_HEADER_KEY = "X-Amzn-Trace-Id"
```

Comment:
Just wondering if these can be defined outside of the class, so an instance of `AwsXRayFormat` doesn't always have to be defined.

Reply:
Awesome! I changed it to be like you mentioned and all tests work and my testings with it worked as well :) Thanks!